easy_access/maintenance/file_existence.py

```
@@ -168,18 +170,22 @@ async def select_items_needing_file_check(
-            """
-        )
+        async for session in get_session():
+            result = await session.execute(
+                text(
+                    f"""
+                SELECT material_id, url
+                FROM copyright_data
+                WHERE {where_clause} AND url IS NOT NULL AND url != ''
+                ORDER BY last_canvas_check ASC NULLS FIRST
+                LIMIT {batch_size} {offset_clause}
+                """
```

Ruff (0.14.6)

Possible SQL injection vector through string-based query construction

(S608)